src/interceptor/interceptor.ts

```diff
@@ -4,8 +4,10 @@ interface TypedInterceptor<T> {
   getPromises: () => Array<any>;
   promise: (fn: () => Promise<any>) => void;
   resolve: () => Promise<any>;
-  on<K extends keyof T>(s: K, props: (props: T[K]) => T[K]);
-  sync<K extends keyof T>(s: K, props: T[K]): T[K];
+  on<K extends keyof T>(key: K, fn: (props: T[K]) => T[K]);
+  sync<K extends keyof T>(key: K, props: T[K]): T[K];
+  awaitForSync<K extends keyof T>(key: K, props: T[K]): Promise<T[K]>;
```

Comment:
i am having hard time understanding the terminology here ;-)
`awaitForSync` but you are you using an async function. `waitFor`  

And both functions return a promise, that's a bit misleading.

Reply:
awaitForSync must return a Promise, otherwise we can't await the end result. And each callback must return a Promise, otherwise we can't awair their single result (in awaitForSync). Basically, I would have said that we could also just change sync and on to always do their work Promise based and the callbacks to always be async functions. Then we get rid of all problems and have a concise API

Reply:
Done